src/gfx/pal_spec.cpp

```diff
@@ -412,6 +421,29 @@ void parseACOFile(std::filebuf &file) {
 	// `codecvt` can be used to convert from UTF-16 to UTF-8
 }
 
+static void parseGBCFile(std::filebuf &file) {
+	// This only needs to be able to read back files generated by `rgbgfx -p`
```

Comment:
Isn't `rgbgfx -p` capable of generating multi-palette output with `-p`? I thought that was how it could be used in place of superfamiconv or Tilemap Studio, for converting whole-screen mockups into a tileset, tilemap, attrmap (assigning palettes 0 to 7), and gbcpal (defining up to 8 palettes of 4 colors each).

Reply:
Up to 256 palettes\*.

Yes, this is multi-palette output, and input again. Why?

Reply:
For one use case, you may do `rgbgfx -o foo.2bpp -p foo.gbcpal foo.png` for one image which uses multiple palettes, and then want to guarantee those same palettes for a paired image `rgbgfx -c gbc:foo.gbcpal -o bar.2bpp bar.png`.

Reply:
Yes, but this is multi-palette input, so what's the problem?

Reply:
The comment says "This only needs to be able to read back files generated by `rgbgfx -p`", and reads into a `char buf[2 * 4]` large enough for only 4 colors. Shouldn't it support as many palettes as the other input format choices?

Reply:
Someone missed the `for (;;)` :stuck_out_tongue:

Reply:
Oh. :P

Can `buf` be scoped inside the `for` then, and rely on any decent optimizer to not reallocate and deallocate stack space for it on every loop?